data_safe_haven/infrastructure/programs/imperative_shm.py

```diff
@@ -172,6 +172,12 @@ def teardown(self) -> None:
             DataSafeHavenAzureError if any resources cannot be destroyed
         """
         logger = get_logger()
+        pulumi_config = DSHPulumiConfig.from_remote(self.context)
```

Comment:
Do we need to handle the case where the pulumi config doesn't exist here?

Reply:
Yes, although maybe in a different PR? At the moment you can't teardown an SHM if no SREs have been deployed, since there is no pulumi config

Reply:
Yes, see updated comment!